src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -765,6 +765,7 @@ public function expects_closer( $node = null ): ?bool {
 		}
 
 		return ! (
+			( $node->has_self_closing_flag ?? false ) ||
```

Comment:
Interestingly, @sirreal, this is no longer required, since we moved to processing the event queue. I'm leaving it in for now, though, while pondering the later change which avoids immediately popping the FORM off the stack of open elements.